lib/pipeline.js

```diff
@@ -291,43 +332,68 @@ class PipelineModel extends BaseModel {
                     .then((existingJobs) => {
                         const jobsToSync = [];
                         const jobsProcessed = [];
+                        const triggersToSync = [];
                         const parsedConfigJobNames = Object.keys(parsedConfig.jobs);
+                        const pipelineId = this.id;
+                        let requiresList = [];
```

Comment:
I think it's better to move let requiresList = []; inside existingJobs.forEach otherwise this can cause problem. E.g. for jobA requiresList = permutations[0].requires, jobB is not in the yaml. Then for jobA it got the requiresList from A since it's not reset.